cmd/clusterctl/examples/aws/generate-yaml.sh

```diff
@@ -65,21 +62,13 @@ if [ $OVERWRITE -ne 1 ] && [ -f $CLUSTER_GENERATED_FILE ]; then
   exit 1
 fi
 
-if [ $OVERWRITE -ne 1 ] && [ -f $PROVIDERCOMPONENT_GENERATED_FILE ]; then
-  echo File $PROVIDERCOMPONENT_GENERATED_FILE already exists. Delete it manually before running this script.
-  exit 1
-fi
-
 mkdir -p ${OUTPUT_DIR}
 
-envsubst < $PROVIDERCOMPONENT_TEMPLATE_FILE \
-  > "${PROVIDERCOMPONENT_GENERATED_FILE}"
-echo "Done generating ${PROVIDERCOMPONENT_GENERATED_FILE}"
-
-envsubst < $CLUSTER_TEMPLATE_FILE \
-  > "${CLUSTER_GENERATED_FILE}"
+envsubst < $CLUSTER_TEMPLATE_FILE > "${CLUSTER_GENERATED_FILE}"
```

Comment:
Could we accomplish this by generating a kustomize patch instead of using envsubst directly against the cluster and machine yaml? It seems odd to modify templates in two separate ways. It may also be possible to use the `kustomize edit` command to modify settings based on env vars directly.

Reply:
I don't think kustomize is there yet to allow for environment substitution, but I might be wrong. We should revisit this as kustomize adds in more features.